docs/java/how-to-develop-a-workflow-definition-in-java.md

```diff
@@ -50,19 +47,93 @@ public interface FileProcessingWorkflow {
+
+**@WorkflowMethod** denotes the starting point of [Workflow Execution](/docs/concepts/what-is-a-workflow-execution). Workflow Execution completes when this method completes.
+A Workflow interface can define only a single method annotated with `@WorkflowMethod`.
 The `@WorkflowMethod` annotation has a `name` parameter, such as `@WorkflowMethod(name = "MyWorkflowType")`.
-It can be used to denote the [Workflow Type](/docs/concepts/what-is-a-workflow-type). If not set, the Workflow Type defaults to the short name of the Workflow interface. In the example above, the Workflow Type defaults to `FileProcessingWorkflow`.
+It can be used to denote the [Workflow Type](/docs/concepts/what-is-a-workflow-type). If not set, the Workflow Type defaults to the short name of the Workflow interface. In the previous example, the Workflow Type defaults to `FileProcessingWorkflow`.
 A method annotated with `@WorkflowMethod` can have any number of parameters. We recommend passing a single parameter that contains all the input fields. This allows adding fields in a backward compatible manner.
+In the following Workflow interface example, the Workflow Type is "NotifyUserAccounts".
+
+```java
+  @WorkflowInterface
+
+  public interface NotifyUserAccounts {
+    @WorkflowMethod
+    void notify(String[] accountIds);
+
+    @QueryMethod
+    int getCount();
+}
+
+```
+
+To overwrite this default naming and assign a custom Workflow Type, use the `@WorkflowMethod` annotation with the `name` parameter, as shown in the following example:
+
+```java
+  @WorkflowMethod(name = "Abc")
+  void notify(String[] accountIds)
+
+```
+
+In this case the Workflow Type will be “Abc”. When you set a Workflow Type in this way, it does not have to start with an upper case letter.
+You can only register a single, unique Workflow Type with a Worker.
+If you try to register multiple Workflow implementations of the same type, you will get an exception.
+
+### `@QueryMethod`
+
+`@QueryMethod` indicates that this method can be used to send a [Query](/docs/concepts/what-is-a-query) to the Workflow's state at any time during its execution.
+It can have parameters that can be used to filter a subset of the Workflow's state that it returns.
+Because it returns a value, it must have a return type that is not `void`.
+A Workflow interface can define any number of methods annotated with `@QueryMethod`, but the method names or the `name` parameters for each must be unique.
 
 The `@QueryMethod` annotation also has a `name` parameter, such as `@QueryMethod(name = "history")`.
 It can be used to denote the Query name.
 If not set, the Query name defaults to the name of the method.
-In the preceding example, the Query name defaults to `getStatus`.
+In the following example, the Query name defaults to `getStatus`.
+
+```java
+@WorkflowInterface
+public interface FileProcessingWorkflow {
+
+    @QueryMethod(name="history")
+    List<String> getHistory();
+
+    @QueryMethod
+    String getStatus();
+}
+```
```

Comment:
I think we can make each annotation example look consistent and clean - pretty sure each one accepts a name parameter?

Reply:
agreed and updated.